charts/generic-service/templates/retry-dlq-cronjob.yaml

```diff
@@ -0,0 +1,33 @@
+{{- if .Values.retryDlqCronjob.enabled -}}
+apiVersion: batch/v1
+kind: CronJob
+metadata:
+  name: {{ include "generic-service.fullname" . | trunc 26 }}-queue-housekeeping-cronjob
+  labels:
+    {{- include "generic-service.labels" . | nindent 4 }}
+spec:
+  {{- if .Values.scheduledDowntime.enabled }}
+  schedule: "{{ .Values.scheduledDowntime.retryDlqSchedule }}"
+  {{- else }}
+  schedule: "{{ .Values.retryDlqCronjob.retryDlqSchedule }}"
+  {{- end }}
+  concurrencyPolicy: Forbid
+  failedJobsHistoryLimit: 5
+  startingDeadlineSeconds: 600
+  successfulJobsHistoryLimit: 5
+  jobTemplate:
+    spec:
+      template:
+        spec:
+          {{- if .Values.serviceAccountName }}
+          serviceAccountName: {{ .Values.serviceAccountName }}
```

Comment:
odd to be using the service account to make a `curl` request to a running container?

Reply:
if you use ` {{ .Values.serviceAccountName }}` then you may give this pod a lot more privileges than it requires.  You could create another value `{{ .Values. retryDlqCronjob.serviceAccountName }}`